modules/javafx.graphics/src/test/java/test/javafx/scene/transform/TransformOperationsTest.java

```diff
@@ -2407,89 +2933,117 @@ private void assertCol3D(MatrixType type, int col, double[] tmp,
-        assertCol3D(MatrixType.MT_3D_3x4, 1, new double[3], true, true, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 2, null, false, false, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 2, null, true, false, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 2, new double[2], true, false, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 2, new double[3], true, true, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 3, null, false, false, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 3, null, true, false, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 3, new double[2], true, false, false);
-        assertCol3D(MatrixType.MT_3D_3x4, 3, new double[3], true, true, false);
-        assertCol3D(MatrixType.MT_3D_3x4, -1, null, true, false, true);
-        assertCol3D(MatrixType.MT_3D_3x4, 4, null, false, false, true);
-
-        assertCol3D(MatrixType.MT_3D_4x4, 0, null, false, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 0, null, true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 0, new double[3], true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 0, new double[4], true, true, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 1, null, false, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 1, null, true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 1, new double[3], true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 1, new double[4], true, true, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 2, null, false, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 2, null, true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 2, new double[3], true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 2, new double[4], true, true, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 3, null, false, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 3, null, true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 3, new double[3], true, false, false);
-        assertCol3D(MatrixType.MT_3D_4x4, 3, new double[4], true, true, false);
-        assertCol3D(MatrixType.MT_3D_4x4, -1, null, true, false, true);
-        assertCol3D(MatrixType.MT_3D_4x4, 4, null, false, false, true);
-    }
-
-    @Test(expected=NullPointerException.class)
-    public void testColumnNullType1() {
-        t.column(null, 0);
-    }
-
-    @Test(expected=NullPointerException.class)
-    public void testColumnNullType2() {
-        t.column(null, 0, new double[] {});
-    }
-
-    @Test
-    public void testSetOnTransformChanged() {
+    @ParameterizedTest
+    @MethodSource("getParams")
+    public void testColumn(Transform t,
+                           Transform it,
+                           boolean is2d,
+                           boolean isIdentity,
+                           boolean isInvertible,
+                           Class inverseType) {
+        assertCol2D(t, is2d, MatrixType.MT_2D_2x3, 0, null, false, false, false);
```

Comment:
the trick with assigning instance fields might have helped here.
I discovered this too late myself, and now it's probably too late anyway...

Reply:
It is either adding a `setUp(...)` line at the start of every function, or editing these. Hard to say which path is really better, so I'd stay with this one.